src/array/uvm_array_op.h

```diff
@@ -0,0 +1,23 @@
+/*!
+ *  Copyright (c) 2019 by Contributors
+ * \file array/array_op.h
+ * \brief Array operator templates
+ */
+#ifndef DGL_ARRAY_UVM_ARRAY_OP_H_
+#define DGL_ARRAY_UVM_ARRAY_OP_H_
+
+#include <dgl/array.h>
+#include <utility>
+
+namespace dgl {
+namespace aten {
+namespace impl {
+
+template <typename DType, typename IdType>
+NDArray IndexSelectCPUFromGPU(NDArray array, IdArray index);
```

Comment:
Is this `IndexSelectCPUToGPU` or `IndexSelectGPUFromCPU`?

Reply:
So the rule I've made here is that if I say FuncXfromY, then X is the memory location and Y is the operator (or processor). I think you mean from the perspective of the actual data movement direction. If you think `IndexSelectCPUToGPU` is easier to understand, we can definitely modify to it.

Reply:
We only need to document it clearly. 
I think IndexSelectCPUFromGPU means selecting data in CPU according to the Index from GPU and return the GPU NDArray.
Can you Add a doc string and also check the context of array and index using assertions?